packages/lit-html/src/lit-html.ts

```diff
@@ -406,9 +407,25 @@ let sanitizerFactoryInternal: SanitizerFactory = noopSanitizer;
  * change in future pre-releases.
  */
 export abstract class Directive {
+  //@internal
+  _part: NodePart | AttributePart;
+  //@internal
+  _attributeIndex: number | undefined;
+  //@internal
+  _directive?: Directive;
+  constructor(part: PartInfo, attributeIndex: number | undefined) {
+    this._part = part as Part;
+    this._attributeIndex = attributeIndex;
+  }
   /** @internal */
-  _resolve(part: Part, props: Array<unknown>): unknown {
-    return this.update(part, props);
+  _resolve(props: Array<unknown>): unknown {
```

Comment:
Remove `_resolve`, can just overwrite `update` on server because we don't need the user `update`.

Reply:
Ah, I remember now: We rely on the base `AttributePart:_setValue` (with `noCommit` argument set to `true`) to resolve the individual directives and return the result on the server. Thus we don't have direct access to the individual directive instances to monkey patch `update` to `render`, and we can't change `Directive.prototype.update` because users override it.

Reply:
I did fix nested directives in SSR and added test for that.